### DISCUSSION TITLE
In future/or now is possible upload files directly from browser to configured S3 storage?

### DISCUSSION THREAD

**Discussion by maxneo4** [pocketbase/pocketbase#1522] [Q&A]
I'm testing the upload files to wasabi S3 provider. It works perfect.

But my azure web app F1 has limited memory usage of 1Gb every certain memory, so the upload limit of file is limited by memory usage. Obvious I don't want to pay money in azure plan with more memory. 

---
**So I was testing for uploading files directly to wasabi using aws-sdk in browser**
I'm thinking to upload directly from client browser. In this way I don't spend memory of my webApp wiht de uploaded files. So I need to link now S3 storage link in one collection's column. it will nice if these were native in js of pocket base.

My question is if is possible pocketbase io js Api support "client browser" upload?, then save the file link to S3 storage like do currently, without the need to use server ram memory. That will be useful when we set pocket base deployed in some cloud provider .

**Comment by ganigeorgiev** [comment]
> My question is if is possible pocketbase io js Api support "client browser" upload?

I'm sorry, I don't understand your request.

Direct client-side upload to external/3rd party storage is up to the developers and there are no plans to integrate it inside the SDKs (_I'm not sure how the security rules will work in this case unless the S3 upload apis are always public or there is some mechanism to retrieve a signed url?_).

**Reply by maxneo4** [reply]
yes, security is an issue because access-key and secret-key will be in js code, at least that you can bring in a encrypted way from pocket base server, so the front could make js upload to S3 storage provider.

I have the doubt if I could set public the file for download directly in Wasabi(using aws-sdk only) or I will need to implement download also in Js.

I like the way pocketbase io manage the uploaded files in S3 storage, but if I upload by my self in JS, so pocketbase won't be able to list files in its API.

By the moment I think I will manage all upload/donwload in the browser, in the pocketBase collection I will store refer to S3 storage file uploaded.
